### PR TITLE
🌱 Use errors.Is() to silence lint warnings

### PR DESCRIPTION
**Pull request by bryanv**
**What does this PR do, and why is it needed?**

In case these become wrapped errors later.

**Which issue(s) is/are addressed by this PR?** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:

Fixes #

**Are there any special notes for your reviewer**:

**Please add a release note if necessary**:

```release-note
NONE
```

### CODE REVIEW

**Comment by akutz** [comment]
I'm not opposed to this, but what IDE is giving these errors? VSCode does not, and our lint target does not. I wonder if we can do something in our linter to catch these. I agree we should use `errors.Is` (or `.As` if we need the value) in case things are ever wrapped.

File: .golangci.yml [hunk 1]
```diff
@@ -143,11 +143,16 @@ linters-settings:
           desc: "replaced by internal packages like pkg/util/ptr"
         - pkg: github.com/onsi/ginkgo$
           desc: "replaced by github.com/onsi/ginkgo/v2"
-
+  errorlint:
+    # Check whether fmt.Errorf uses the %w verb for formatting errors.
```
Comment (akutz) [inline .golangci.yml:147]: I _want_ `Errorf` to use `%w` to format errors. Why would we wish to disable that?

Comment (bryanv) [reply .golangci.yml:147]: That disables the lint for not using %w. There's lots of places where we're not and something that can be addressed later (along with the other disabled check)

**Review by akutz — APPROVED** [review]
Sorry for not getting back to this.

**Pull request merged**